tests/end_to_end/end_to_end.cpp

```diff
@@ -43,6 +46,76 @@ prepare_udp_packet(uint16_t udp_length)
     return packet;
 }
 
+const std::array<uint8_t, 6> _test_source_mac = {0, 1, 2, 3, 4, 5};
+const std::array<uint8_t, 6> _test_destination_mac = {0xa, 0xb, 0xc, 0xd, 0xe, 0xf};
+const in_addr _test_source_ip = {10, 0, 0, 1};
+const in_addr _test_destination_ip = {20, 0, 0, 1};
+
+typedef class _udp_packet
+{
+  public:
+    _udp_packet(
+        _In_ const std::array<uint8_t, 6>& source_mac = _test_source_mac,
+        _In_ const std::array<uint8_t, 6> destination_mac = _test_destination_mac,
+        _In_ const in_addr* source_ip = &_test_source_ip,
+        _In_ const in_addr* destination_ip = &_test_destination_ip,
```

Comment:
Don't use version-specific types with version-agnostic names.
E.g., "udp_packet" as a class implies both IPv4 and IPv6, as does "source_ip" and "destination_ip"